test/blackbox/run_blackbox_tests/run_blackbox_tests.c

```diff
@@ -0,0 +1,133 @@
+/*
+    run_blackbox_tests.c -- Implementation of Black Box Test Execution for meshlink
+
+    Copyright (C) 2017  Guus Sliepen <guus@meshlink.io>
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+#include <stdlib.h>
+#include <stdarg.h>
+#include <setjmp.h>
+#include <cmocka.h>
+#include <assert.h>
+#include "execute_tests.h"
+#include "test_cases.h"
+#include "test_cases_open.h"
+#include "test_cases_start.h"
+#include "test_cases_stop_close.h"
+#include "test_cases_send.h"
+#include "test_cases_pmtu.h"
+#include "test_cases_get_self.h"
+#include "test_cases_get_node.h"
+#include "test_cases_add_addr.h"
+#include "test_cases_get_ex_addr.h"
+#include "test_cases_add_ex_addr.h"
+#include "test_cases_get_port.h"
+#include "test_cases_blacklist.h"
+#include "test_cases_default_blacklist.h"
+#include "test_cases_whitelist.h"
+#include "test_cases_channel_open.h"
+#include "test_cases_channel_close.h"
+#include "test_cases_channel_send.h"
+#include "test_cases_channel_shutdown.h"
+
+#include "test_cases_destroy.h"
+#include "test_cases_get_all_nodes.h"
+#include "test_cases_get_fingerprint.h"
+#include "test_cases_rec_cb.h"
+#include "test_cases_sign.h"
+#include "test_cases_set_port.h"
+#include "test_cases_verify.h"
+#include "test_cases_invite.h"
+#include "test_cases_export.h"
+#include "test_cases_channel_ex.h"
+#include "test_cases_channel_get_flags.h"
+#include "test_cases_status_cb.h"
+#include "test_cases_set_log_cb.h"
+#include "test_cases_join.h"
+#include "test_cases_import.h"
+#include "test_cases_channel_set_accept_cb.h"
+#include "test_cases_channel_set_poll_cb.h"
+#include "test_cases_channel_set_receive_cb.h"
+#include "test_cases_hint_address.h"
+#include "../common/containers.h"
+#include "../common/common_handlers.h"
+
+#define CMD_LINE_ARG_MESHLINK_ROOT_PATH 1
+#define CMD_LINE_ARG_LXC_PATH 2
+#define CMD_LINE_ARG_LXC_BRIDGE_NAME 3
+#define CMD_LINE_ARG_ETH_IF_NAME 4
+#define CMD_LINE_ARG_CHOOSE_ARCH 5
+
+char *meshlink_root_path = NULL;
+char *choose_arch = NULL;
+int total_tests;
+
+int main(int argc, char *argv[]) {
+	/* Set configuration */
+	assert(argc >= (CMD_LINE_ARG_CHOOSE_ARCH + 1));
+	meshlink_root_path = argv[CMD_LINE_ARG_MESHLINK_ROOT_PATH];
```

Comment:
I would get rid of these `CMD_LINE_ARG_*` macros. Just write:

    meshlink_root_path = argv[1];
    lxc_path = argv[2];
    ...

That's easier to read, and not less safe at all. Of course, magic numbers should be avoided, but magic numbers are only magic if it's unclear from the context what they mean. In the case of `argv[]`, it's immediately clear that `argv[1]` means the first command line argument. Also, since you immediately assign it to a variable here, it's also clear that the first command line argument is the root path.

The same goes for the other files where this is done.